torchtext/models/roberta/bundler.py

```diff
@@ -156,10 +155,13 @@ def encoderConf(self) -> RobertaEncoderConf:
 XLMR_BASE_ENCODER = RobertaModelBundle(
     _path=urljoin(_TEXT_BUCKET, "xlmr.base.encoder.pt"),
     _encoder_conf=RobertaEncoderConf(vocab_size=250002),
-    transform=partial(get_xlmr_transform,
-                      vocab_path=urljoin(_TEXT_BUCKET, "xlmr.vocab.pt"),
-                      spm_model_path=urljoin(_TEXT_BUCKET, "xlmr.sentencepiece.bpe.model"),
-                      )
+    transform=lambda: T.Sequential(
+        T.SentencePieceTokenizer(urljoin(_TEXT_BUCKET, "xlmr.sentencepiece.bpe.model")),
+        T.VocabTransform(load_state_dict_from_url(urljoin(_TEXT_BUCKET, "xlmr.vocab.pt"))),
+        T.Truncate(510),
+        T.AddToken(token=0, begin=True),
+        T.AddToken(token=2, begin=False),
```

Comment:
Is it guaranteed that the `bos` and `eos` tokens will always have an index of 0 and 2 respectively? Could there be a usecase where the user uses their own vocab instead of loading from `xlmr.vocab.pt`?

Reply:
Certainly plausible that they'd add, say, additional special tokens. They would be appended to the vocab though, so 0 and 2 should remain as BOS and EOS respectively. If they didn't, you'd break the "contract" with how RoBERTa was trained.

Reply:
These are pre-trained models, and the transform comply with what the model was originally trained with. Certainly users can bring in their own flavors or Sentencepiece models, vocab etc, but then that transform is specific to user's model and won't apply to the pre-trained weights.

Reply:
Got it thanks for the clarification!